### PR TITLE
Signatures: Make module testable, add test for hashed method

### PR DESCRIPTION
**Pull request by jzaefferer**
In preparation for upcoming changes to the signatures module.

Ref #32

I'm not too happy with this approach, but I find it still much better over mocking `http.request`.

### CODE REVIEW

File: lib/signatures.js [hunk 1]
```diff
@@ -154,5 +154,7 @@ module.exports = {
 	caa: caa,
 	all: all,
 	hashed: hashed,
+	// only for tests
```
Comment (scottgonzalez) [inline lib/signatures.js]: Spacing and capitalization.

**Comment by scottgonzalez** [comment]
Seems ok.

**Comment by jzaefferer** [comment]
Fixed the line comment.

File: test/signatures-caa.json [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+			"$t": "Alexander Schmitz"
+		},
+		"gsx$emailaddress": {
+			"$t": "arschmitz@gmail.com"
+		},
+		"gsx$confirmation": {
+			"$t": "I AGREE"
+		}
+	}
+]
```
Comment (arschmitz) [inline test/signatures-caa.json]: Missing new line

File: test/signatures-caa.json [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+		"gsx$fullname": {
+			"$t": "Alexander Schmitz"
+		},
+		"gsx$emailaddress": {
+			"$t": "arschmitz@gmail.com"
+		},
+		"gsx$confirmation": {
+			"$t": "I AGREE"
+		}
+	}
```
Comment (arschmitz) [inline test/signatures-caa.json]: Might consider adding yourself the `ü` tends to cause issues in general if not properly handled. So would be good to have that in the tests.

File: test/signatures-cla.json [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+			"$t": "bobbelcher@gmail.com"
+		},
+		"gsx$confirmation": {
+			"$t": "I AGREE"
+		},
+		"gsx$nameconfirmation": {
+			"$t": ""
+		}
+	}
+]
```
Comment (arschmitz) [inline test/signatures-cla.json]: Missing new line

**Comment by jzaefferer** [comment]
Addressed all three, despite the comments not getting marked as outdated.

**Comment by arschmitz** [comment]
looks good to me :+1:

**Pull request merged**